[V3 General] Fix error in [p]userinfo when game is not set

skmendez opened the pull request:
### Type

- [X] Bugfix
- [ ] Enhancement
- [ ] New feature

### Description of the changes

Modifies the game-checking lines so that they don't assume `user.game` is not a NoneType object.

Fixes #1321

skmendez committed:
Fix #1321

Modifies the game-checking lines so that they don't assume `user.game` is not a NoneType object.

tekulvw reviewed (changes requested):
There is a possibility of `game` not being set at all. I'd recommend the following:

```py
game = _("Chilling in {} status").format(user.status)
if user.game is None:
    pass
...
```

skmendez commented:
Clarify what you mean by `game` not being set at all. For all `Member` objects, they have an attribute `game` (which may or may not be `None`). The only condition where my solution would error is if there was a `user.game.type not in [0, 1, 2, 3]`

skmendez committed:
Fix #1321

Modifies the game-checking lines so that they don't assume user.game is not a NoneType object. Also alerts the user to go bother @tekulvw if somehow the invariants of `user` having a `game` attribute and the `game` attribute either being `None` of having a `type` attribute in `[0, 1, 2, 3]` are broken.

Tobotimus commented on redbot/cogs/general/general.py:
```diff
@@ -216,17 +216,19 @@ def __init__(self):
-
-        # Check if the user has a special status
-        if user.game.type == 0:  # "Playing" status
+        
```
Bit of trailing whitespace here on 219 could be removed

Tobotimus commented on redbot/cogs/general/general.py:
```diff
@@ -216,17 +216,19 @@ def __init__(self):
         elif user.game.type == 3:  # "Watching" status
             game = _("Watching {}").format(user.game.name)
+        else:
+            game = "please go bother <@111655405708455936> as somehow something broke"
```
I prefer Will's solution which sends the default message if `user.game` is None or something we don't expect.

Tobotimus replied:
Wait a second, don't remove the `else` block. I'd rather have it throw an error (like a `ValueError` or `RuntimeError`) or just display the default "chilling" message as Will's example provides.

Tobotimus reviewed (changes requested):
Otherwise I think this is fine

skmendez committed:
Update general.py

Tobotimus reviewed (approved):
Will wait on @tekulvw for the final review
